snapcraft/internal/project_loader/_extensions/gnome_3_34.py

```diff
@@ -88,6 +88,7 @@ def __init__(self, *, extension_name: str, yaml_data: Dict[str, Any]) -> None:
             "environment": {
                 "SNAP_DESKTOP_RUNTIME": "$SNAP/gnome-platform",
                 "GTK_USE_PORTALS": "1",
+                "PYTHONPATH": "$PYTHONPATH:$SNAP/gnome-platform/usr/lib/python3.6/site-packages",
```

Comment:
This is a well used variable which might make more sense to have in the command-chain instead or will override the users input. There's also a slight issue here, there is no prior PYTHONPATH defined in the environment, so the actual python path might end up being ".:$SNAP/gnome-platform/usr/lib/python3.6/site-packages" (if the LD_LIBRARY_PATH behavior applies to PYTHONPATH)

Reply:
Good points @sergiusens .. let's explore this..

> The behavior could be that if PYTHONPATH is -z...

I don't quite get your meaning with this -z parameter. Can you elaborate and maybe provide an example of what you're thinking?

> This is a well used variable which might make more sense to have in the command-chain instead or will override the users input.

If the user set the `PYTHONPATH`, then wouldn't that just be the `$PYTHONPATH` going into this environment variable setting? So that the `PYTHONPATH` would end up being `<pythonpath from user setting>:$SNAP/gnome-platform/usr/lib/python3.6/site-packages`?

> so the actual python path might end up being ".:$SNAP/gnome-platform/usr/lib/python3.6/site-packages" 

What problems does this cause? I suppose we could swap the order so that `$PYTHONPATH` is at the end but wouldn't the behavior be the same?

Reply:
we had a chat about it and decided it was best if PYTHONPATH lives in launcher-specific.